examples/reward_modeling/train_reward_model.py

```diff
@@ -0,0 +1,169 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass, field
+from typing import Optional
+
+from datasets import load_dataset
+from peft import LoraConfig
+from tqdm import tqdm
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoTokenizer,
+    BitsAndBytesConfig,
+    HfArgumentParser,
+    TrainingArguments,
+)
+
+from trl import RewardTrainer
+
+
+tqdm.pandas()
+
+########################################################################
+# This is a fully working simple example to use RewardTrainer from trl on any HF model.
+#
+# This example fine-tunes OPT model on the hh-rlhf dataset
+# We eventually use adapters from PEFT library to train Low Rank Adapters.
+# in any of the following settings (with the same script):
+#   - single CPU or single GPU
+#   - multi GPUS (using PyTorch distributed mode)
+#   - multi GPUS (using DeepSpeed ZeRO-Offload stages 1 & 2)
+#   - fp16 (mixed-precision) or fp32 (normal precision)
+#
+# To run it in each of these various modes, first initialize the accelerate
+# configuration with `accelerate config`
+#
+########################################################################
+
+########################################################################
+# NOTE for to train with a 8-bit model a more recent version of
+# transformers is required, full dependecies for this example:
+# pip install bitsandbytes datasets accelerate
+# pip install git+https://github.com/huggingface/peft.git
+########################################################################
```

Comment:
I would just mention this in the readme? Feels a bit an odd place for documentation. what do you think?

Reply:
Agreed !